apps/test-app/frontend/src/components/tree-widget/TreeWidget.tsx

```diff
@@ -273,3 +336,56 @@ function parseTreeNodeItem(node: HierarchyNode): DelayLoadedTreeNodeItem {
     autoExpand: node.autoExpand,
   } as DelayLoadedTreeNodeItem;
 }
+
+function createInfoNode(parentNode: TreeNodeItem | undefined, message: string, type?: InfoTreeNodeItemType): PresentationInfoTreeNodeItem {
+  const id = parentNode ? `${parentNode.id}/info-node` : `/info-node/${message}`;
+  return {
+    id,
+    parentId: parentNode?.id,
+    label: PropertyRecord.fromString(message),
+    message,
+    isSelectionDisabled: true,
+    children: undefined,
+    type: type ?? InfoTreeNodeItemType.Unset,
+  };
+}
+
+interface StatelessTreeNodeRendererProps extends TreeNodeRendererProps {
+  onLimitReset: (parentId?: string) => void;
+}
+
+function StatelessTreeNodeRenderer(props: StatelessTreeNodeRendererProps) {
+  const nodeItem = props.node.item;
+  if (isPresentationInfoTreeNodeItem(nodeItem)) {
+    return (
+      <TreeNode
+        isLeaf={true}
+        label={
+          <span>
+            <Text isMuted>
+              <span>{nodeItem.message}</span>
+              {nodeItem.type === InfoTreeNodeItemType.ResultSetTooLarge && (
+                <span>
+                  <span> - </span>
+                  <UnderlinedButton
+                    onClick={() => {
+                      const parsedParentIds = nodeItem.parentId ? JSON.parse(nodeItem.parentId) : undefined;
+                      const parentId = parsedParentIds ? parsedParentIds[parsedParentIds.length - 1].instanceKeys[0].id : undefined;
+                      props.onLimitReset(parentId);
```

Comment:
can you explain what's going on here?

Reply:
`nodeItem.parentId` contains values like:
```json
"[{\"type\":\"instances\",\"instanceKeys\":[{\"className\":\"BisCore.Subject\",\"id\":\"0x1\"}]},{\"type\":\"instances\",\"instanceKeys\":[{\"className\":\"BisCore.PhysicalModel\",\"id\":\"0x1c\"}]},{\"type\":\"instances\",\"instanceKeys\":[{\"className\":\"BisCore.SpatialCategory\",\"id\":\"0x17\"}]}]"
```
I was parsing it and extracting only the last parent Id (would be "0x17" in the example above). Then we were changing map of key(parent id): value (level limit) pairs. This map was later used before running the `getNodes(...)` to extract limit of specific node.

Reply:
Changed the implementation to not parse parent id, but instead use the exact `parentId` in `[parentId: string]: number | "unbounded" | undefined}`. Now `ChildNodesCache` doesn't include limit into the `key` if `parentNode` is a grouping node.

Reply:
Should I document that `getNodes(...)` function doesn't use provided limit when `parentNode` is a `GroupingNode`?

Reply:
Looks good. As discussed, let's also change `GetHierarchyNodesProps` to not have `instanceFilter` and `hierarchyLevelSizeLimit` attributes for grouping nodes.

Reply:
Tried doing that, but as discussed keeping as is. Added a comment for `instanceFilter` and `hierarchyLevelSizeLimit`: 
```
Isn't applied if `parentNode` is a [[GroupingNode]].
```